lib/src/ui/pages/presentation/presentation_page.dart

```diff
@@ -0,0 +1,195 @@
+                _navigateToHomePage(context);
+              } else {
+                _goNextStep();
+              }
+            },
+          ),
+          const Spacer(),
+          FlatButton(
+            onPressed: () => _navigateToHomePage(context),
+            child: Text(Strings.of(context).skip),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget _buildPresentationContent() => Expanded(
+      child: PageView.builder(
+          physics: const PageScrollPhysics(),
+          controller: _controller,
+          itemBuilder: (BuildContext context, int index) {
+            final presentationSteps = _presentationStepsProvider.provide(context);
+            return (index < presentationSteps.length) ? _buildPresentationStep(context, presentationSteps[index]) : null;
+          }));
+
+  Widget _buildPresentationStep(BuildContext context, PresentationStep presentationStep) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        Expanded(child: presentationStep.presentationImageType.getImage()),
+        Column(
+          children: [
+            const SizedBox(height: _Constants.presentationStepVerticalSpacing),
+            Text(
+              presentationStep.title,
+              style: Theme.of(context).textTheme.bodyText1.copyWith(fontWeight: FontWeight.bold),
+            ),
+            const SizedBox(height: _Constants.presentationStepVerticalSpacing),
+            Text(presentationStep.description),
+            const SizedBox(height: _Constants.presentationStepVerticalSpacing),
+          ],
+        )
+      ],
+    );
+  }
+
+  void _goNextStep() {
+    _controller.animateToPage(
+      _controller.page.toInt() + 1,
+      duration: _Constants.indicatorAnimationDuration,
+      curve: _Constants.indicatorAnimationCurve,
+    );
+  }
+
+  void _navigateToHomePage(BuildContext context) {
+    context.read<PresentationNotifier>().completePresentation();
+    Navigator.pushReplacement(
+      context,
+      MaterialPageRoute(
+        builder: (context) => HomePage(),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _controller.dispose();
+    super.dispose();
+  }
+}
+
+class IndicatorContainer extends AnimatedWidget {
+  const IndicatorContainer({
+    @required this.controller,
+    @required this.itemCount,
+    @required this.onPageSelected,
+  }) : super(listenable: controller);
+
+  final PageController controller;
+  final int itemCount;
+  final ValueChanged<int> onPageSelected;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: List<Widget>.generate(itemCount, _buildIndicator),
+    );
+  }
+
+  Widget _buildIndicator(int index) {
+    final double animationValue =
+        Curves.easeOut.transform(max(0, 1.0 - ((controller.hasClients ? controller.page : controller.initialPage) - index).abs()));
```

Comment:
zastanawiam się czy dałoby się to zrobić Tweenem, próbowałeś coś z tym?

Reply:
Przyznam, że nie próbowałem. Generalnie ten PageController dedykowany PageView zawiera wartość określającą stan swipa, tj. jeżeli 2 widoki będą po równo w połowie widoczne na ekranie to wartość będzie 0.5 dla obu, więc powiedzmy, że nie jest potrzebny Tween do interpolowania wartości - druga sprawa, to zauważyłem, że w sumie to ogólnie niepotrzebnie tutaj interpoluje transformacją, bo w sumie ta wartość z pageControllera wystarcza :D